site/docs/blog/incident-2022-05.md

```diff
@@ -0,0 +1,115 @@
+# Nessie issue during rolling upgrade from <=0.25.0 to >=0.26.0
+
+## Symptom
+
+During or after a rolling upgrade from Nessie version <= 0.25.0 to >= 0.26.0, exceptions/errors like
+`org.projectnessie.versioned.ReferenceNotFoundException: Global log entry '<hex>’ not does not exist.`
+and/or `Iceberg content from reference must have global state, but has none`.
+
+## Identification of the issue and mitigation in the live system
+
+(add content - Dimitri)
+
+## Background
+
+When Nessie runs against non-transactional databases, it uses a [“global pointer”](https://github.com/projectnessie/nessie/blob/nessie-0.26.0/versioned/persist/serialize-proto/src/main/proto/persist.proto#L105),
+which holds the mapping of all named references to their HEAD commit IDs, the HEAD of the ref-log
+and the HEAD of the global-log. Every update to the Nessie repository ends in a CAS[^1] on that
+single global pointer. If the CAS is successful, the change, for example a commit or merge
+operation, was atomically & consistently applied.[^2]
+
+The approach to maintain these three HEADs in a single "row" works, but it does not scale well.
+This “single point of contention” was never meant to stay forever, just as long as we need it
+and/or do not have a better solution for it.
+
+We implemented Nessie using this concrete global pointer mechanism, because certain decisions
+haven’t been made at that time, and we wanted to be on the “safe side” and then see what can be
+improved.
+
+## Analysis
+
+Recently we were certain that having the so-called “global state” for Iceberg tables and views is
+actually not such a great thing. So the team decided that the “global state” can go away. This was
+implemented in the Nessie PR #3866. Since that change reduced the amount of global-log-entries to
```

Comment:
#3866 suppose to be a hyper link?

Reply:
>  try it - it does

I clicked on view doc (it opens markdown view), but I cannot see it as a hyperlink.

Reply:
<img width="1724" alt="image" src="https://user-images.githubusercontent.com/5889404/171195886-18f2345a-6437-4272-88fe-f52582b6bfec.png">

Reply:
I think PR hyperlinks work in GH only, not in the Project Nessie docs.

Reply:
`mkdocs serve` show the correct links.